src/gtk/toga_gtk/widgets/numberinput.py

```diff
@@ -5,37 +5,59 @@
 
 class NumberInput(Widget):
     def create(self):
-        adjustment = Gtk.Adjustment(0, self.interface.min_value,
-                                    self.interface.max_value,
-                                    self.interface.step, 10, 0)
-
         self.native = Gtk.SpinButton()
-        self.native.set_adjustment(adjustment)
         self.native.set_numeric(True)
         self.native.interface = self.interface
 
         self.rehint()
 
+    def set_adjustment_value(self):
+        adjustment = Gtk.Adjustment(0, self.interface.min_value,
+                                        self.interface.max_value,
+                                        self.interface.step, 10, 0)
+        self.native.set_adjustment(adjustment)
+
     def set_readonly(self, value):
-        self.native.editable = not value
+        self.native.set_property('editable', not value)
 
     def set_step(self, step):
-        self.interface.factory.not_implemented('NumberInput.set_step()')
+        if step:
+            self.native.set_increments(step,step)
 
     def set_min_value(self, value):
-        self.interface.factory.not_implemented('NumberInput.set_min_value()')
+        #self.native.set_range(min=value)
+        if value:
+            try:
+                self.native.set_range(value,self.interface.max_value)
+            except (AttributeError):
+                pass
```

Comment:
The AttributeError should not be needed here - max_value should always exist. It might not have a numeric value - e.g., it might be None, indicating an open ended interval - but it will always exist.

Reply:
```
  File "/home/yugesh/Documents/beeware/toga/src/gtk/toga_gtk/widgets/numberinput.py", line 30, in set_min_value
    self.native.set_range(value,self.interface.max_value)
TypeError: Argument 2 does not allow None as a value
```

With max_value as None.. 😕

Reply:
nothing in the documentation of Gtk.SpinButton said sanything about having a variable upper limit..